packages/tenable_io/data_stream/scanner/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,190 @@
+---
+description: Pipeline for processing Tenable.io scanner logs.
+processors:
+  - set:
+      field: ecs.version
+      value: '8.7.0'
+  - set:
+      field: event.category
+      value: [host]
+  - set:
+      field: event.type
+      value: [info]
+  - rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+  - json:
+      field: event.original
+      target_field: json
+  - set:
+      field: event.kind
+      value: state
+  - set:
+      field: event.ingested
+      value: '{{_ingest.timestamp}}'
+  - set:
+      field: host.hostname
+      copy_from: json.hostname
+      ignore_empty_value: true
+  - set:
+      field: host.name
+      copy_from: json.name
+      ignore_empty_value: true
+  - set:
+      field: host.ip
+      copy_from: json.ip_addresses
+      ignore_empty_value: true
+  - remove:
+      field: host.ip
+      if: ctx.host?.ip != null && ctx.host?.ip[0] == ""
+  - set:
+      field: host.os.type
+      copy_from: json.platform
+      ignore_empty_value: true
+  - lowercase:
+      field: host.os.type
+      ignore_missing: true
+  - set:
+      field: host.os.full
+      copy_from: json.distro
+      ignore_empty_value: true
+  - script:
+      description: Convert 'json.linked' integer value of '0' or '1' to boolean
+      lang: painless
+      source:
+        if (ctx.json?.linked != null && ctx.json?.linked != '') {
+          if (ctx.json?.linked == 0) {
+            ctx.json.linked = false
+          }
+          if (ctx.json?.linked == 1) {
```

```suggestion
        if (ctx.json?.linked != null && ctx.json.linked != '') {
          if (ctx.json.linked == 0) {
            ctx.json.linked = false
          }
          if (ctx.json.linked == 1) {
```
What are the samantics when `json.linked` is not present? Is that equivalent to `false` or does it have some other meaning?

(similar changes below)

---

The value of `linked` will always be one of `0` or `1`. In the year or more I have been collecting this data via Python, there has not been an empty or null value. I had more or less just put the check as error handling, but I could remove that check if you would prefer and just do:

```
        if (ctx.json.linked == 0) {
          ctx.json.linked = false
        }
        if (ctx.json.linked == 1) {
          ctx.json.linked = true
        }
```

Eric

---

I think we should at least guard a little.

---

My concern is that if it does change, this will result in a mapping error and document loss.

---

We should be good after yesterdays changes, then. I will mark this as resolved, but let me know if you see anything else.